internal/infrastructure/kubernetes/proxy_deployment_test.go

```diff
@@ -152,6 +153,35 @@ func TestExpectedProxyDeployment(t *testing.T) {
 	assert.Equal(t, repl, *deploy.Spec.Replicas)
 }
 
+func TestExpectedBootstrap(t *testing.T) {
+	svrCfg, err := config.New()
+	require.NoError(t, err)
+	cli := fakeclient.NewClientBuilder().WithScheme(envoygateway.GetScheme()).WithObjects().Build()
+	kube := NewInfra(cli, svrCfg)
+	infra := ir.NewInfra()
+
+	infra.Proxy.GetProxyMetadata().Labels[gatewayapi.OwningGatewayNamespaceLabel] = "default"
+	infra.Proxy.GetProxyMetadata().Labels[gatewayapi.OwningGatewayNameLabel] = infra.Proxy.Name
+
+	// Set a custom bootstrap config into EnvoyProxy API and ensure the same
+	// value is set as the container arg.
+	bstrap := "blah"
```

Comment:
This is obvious a invalid bootstrap config I think? There should be an error reported.

Reply:
as I mentioned in the prevous comment, the validation will happen earlier. Just used a dummy string to test in this component.